packages/public/server/__tests-e2e__/_utils.ts

```diff
@@ -20,23 +20,288 @@
   )
+
   return queryResults[0]
 }
+
+export async function goto(site: string): Promise<ElementHandle> {
+  await page.goto(testingServerUrl + site)
+
+  return getDocument(page)
+}
+
+export async function click(element: ElementHandle): Promise<void> {
+  await element.click()
+  await page.waitFor(100)
+}
+
+export async function clickForNewPage(
+  element: ElementHandle
+): Promise<ElementHandle> {
+  await element.click()
+  await page.waitForNavigation({ waitUntil: 'networkidle0' })
+
+  return getDocument(page)
+}
+
+function just<T>(x: T): NonNullable<T> {
+  assert.ok(x !== null)
+
+  return x!
+}
+
+export async function isVisible(
+  element: ElementHandle<HTMLElement>
+): Promise<boolean> {
+  return element.evaluate(
+    e => !!(e && (e.offsetWidth || e.offsetHeight || e.getClientRects().length))
+  )
+}
+
+export async function getText(element: ElementHandle): Promise<string> {
+  return element.evaluate(e => e.textContent).then(just)
+}
+
+export async function getMainContent(
+  element: ElementHandle
+): Promise<ElementHandle> {
+  return getBySelector(element, '#page')
+}
+
+export async function login(user: string): Promise<void> {
+  await logout()
+
+  const loginPage = await goto(pages.login.path)
+
+  const { buttonLogin, inputUser, inputPassword } = pages.login.identifier
+  await getByPlaceholderText(loginPage, inputUser).then(e => e.type(user))
+  await getByPlaceholderText(loginPage, inputPassword).then(e =>
+    e.type(pages.login.defaultPassword)
+  )
+
+  await getByText(loginPage, buttonLogin).then(clickForNewPage)
+}
+
+export async function logout(): Promise<void> {
+  await goto(pages.logout.path)
+}
+
+export async function openDropdownMenu(topic: ElementHandle) {
+  await getBySelector(
+    topic,
+    '#subject-nav-wrapper button.dropdown-toggle'
+  ).then(click)
+  return await page.waitForSelector('#subject-nav-wrapper .dropdown-menu')
+}
+export function addContent(type: string) {
+  return async (topic: ElementHandle) => {
+    await getByText(topic, 'Add content').then(e => e.hover())
+    return await getByText(topic, type).then(clickForNewPage)
+  }
+}
+export async function organizeTaxonomy(topic: ElementHandle) {
+  return await getByText(topic, 'Organize taxonomy').then(clickForNewPage)
+}
+
+export async function saveRevision(createPage: ElementHandle) {
+  await getBySelector(createPage, '#subject-nav-wrapper .fa-save').then(click)
+  await getByLabelText(createPage, 'Änderungen').then(e => e.type(randomText()))
+  await createPage.$$('input[type=checkbox]').then(c => c[0].click())
+  await createPage.$$('input[type=checkbox]').then(c => c[3].click())
+
+  return await getByText(createPage, 'Speichern', {
+    selector: 'button'
+  }).then(clickForNewPage)
+}
+
+export function randomText(prefix?: string): string {
+  return (prefix ?? 'Text') + ' ' + String(Math.floor(Math.random() * 1e12))
+}
+
+export async function toHaveAttribute(
```

Comment:
Instead of export all these, we should have a file `_extend-jest.js` or something like that and call `expect.extend` there. Because now you split the declaration and types into `jest.setup-e2e.ts` and `_utils.ts` (so you need to touch multiple places if you change the signature / add new matchers)